src/helpers.tsx

```diff
@@ -2,28 +2,47 @@
 import { WrapperSettings } from '@storybook/addons';
 import { DEFAULT_PADDING } from './constants';
 
-type Option = {
-  name: string;
-  value: string;
-  default?: boolean;
+type Size = { name: string; value: string };
+
+type Option = Size & {default?: boolean}
+
+type Sizes = { [key: string]: Size };
+
+type PaddingConfig = {
+  sizes: Sizes;
+  defaultPadding: string;
 }
 
-type Options = Option[] | WrapperSettings['parameters'];
+type Options = Option[] | PaddingConfig | WrapperSettings['parameters'];
+
+export const normalizeEntries = (options: Options) => (Array.isArray(options)
+  ? options
+  : Object.entries<Size>(options.sizes).map(([key, { name, value }]) => {
+    const isDefault = options.defaultPadding === key;
 
-export const isEnabled = (options: Options): options is Option[] => options.length > 0;
+    return { name, value, default: isDefault };
+  }));
+
+export const isEnabled = (options: Options) => {
+  const items = Array.isArray(options) ? options : Object.entries(options.sizes);
```

Comment:
Must check the value of `disable` field as well when it's an object. This wasn't needed before because anything other than a non-empty array was considered disabled.